sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveTableScanSuite.scala

```diff
@@ -90,4 +90,11 @@ class HiveTableScanSuite extends HiveComparisonTest {
     assert(sql("select CaseSensitiveColName from spark_4959_2").head() === Row("hi"))
     assert(sql("select casesensitivecolname from spark_4959_2").head() === Row("hi"))
   }
+
+  test("SPARK-8003 Resolve spark__partition__id in Hive queries") {
+    TestHive.sql("CREATE TABLE tb (KEY INT, VALUE STRING)")
```

Comment:
I added this test since you had asked to add the virtual column resolution to `HiveContext` and this tests to make sure that it is being used.The reason I initially placed the test here since it was to see if Hive would resolve the column.